### DISCUSSION TITLE
Let the user select a text anywhere and launch FL search with that selected text without having to press CTRL+C and CTRL+V

### DISCUSSION THREAD

**Discussion by raindropsfromsky** [Flow-Launcher/Flow.Launcher#1179] [Ideas]
Currently, if we see any text in any app (e.g. browser, Word/Excel/PowerPoint/pdf file, etc), and if we want to search that in FL, we have to do it in four distinct steps:

1. Select the text
2. press CTRL+C
3. launch FL by pressing its hotkey (e.g. CTRL+SpaceBar)
4. press CTRL+V

Shorten this process as follows:
1. Select a text anywhere (in any application) 
2. launch FL QuickSearch with a modified hotkey (e.g. CTRL+SHFT+SpaceBar).

We should not need CTRL+C or CTRL+V.

[WordWeb](https://wordweb.info/) has this feature: Select any text in any app and press CTRL+W. That launches WordWeb and loads the selected text in it, and WordWeb shows the meaning of that word. We can also use **CTRL+RMB click** for the same action.

**Comment by taooceros** [comment]
I like this feature, but the implementation is under consideration. I am not sure whether invoking clipboard and read from clipboard is a good implementation, but it is hard to have other implementation.

**Reply by jjw24** [reply]
I know the intention is CTRL + C is also not used, but I think this can be done easily via a plugin after pressing CTRL + C, after which the plugin then access the clipboard and grab the first item upon action hotkey trigger, similar to the Clipboard History plugin. User can map a custom query to this so it will be triggered with shortcut.

If we also not use CTRL + C, I am more curious how to detect the selected text in browser, Word/Excel/PowerPoint/pdf file, etc.

**Comment by raindropsfromsky** [comment]
Thanks for considering!

BTW even the **SearchBar** addon of Chrome has the same feature.
It can take the selected word and pop up any website as a separate tab (or same tab), with that selected text as search string.

But compared to WordWeb, it has a more limited scope of work: It can capture the selected word within the given Chrome window only.
But in our context, I am focusing on the way it can capture the selected text and process it.

**Comment by aghster** [comment]
> Shorten this process as follows:
> 
>  1. Select a text anywhere (in any application)
> 
>  2. launch FL QuickSearch with a modified hotkey (e.g. CTRL+SHFT+SpaceBar).
> 
> 
> We should not need CTRL+C or CTRL+V.

This is exacly what I am looking for too: #1135

**Reply by aurelio20000** [reply]
It would be good to have this feature of pressing control shift and space, after choosing a word or series of words and for FL to immediately send it to the text bar for modification, with the feature of being able to send it to a search engine predetermined by the user, without the need to declare it at every moment

**Comment by zunaidFarouque** [comment]
I want this implemented too!
This is what I was looking for, and I found this issue.
Please add this feature...

It could be like this: (example workflow)
usual `alt+space` = flow launcher default window
unique hotkey like `alt+shift+space` or `alt+win+space` for flow launcher with selected text (flow will trigger ctrl+c in background to copy the selected text to the clipboard, then open the window and the copied text will be there)

### Example use case:
#### 1: JSON creation using AI
Suppose you need to create an example array of 5 JSON objects with the following properties: name, age, and occupation.
You select `Create an example array of 5 JSON objects with the following properties: name, age, and occupation` and press `Alt+Shift+Space`, which will open the Flow Launcher with the text already placed. then you press keyboard `Home` button to get to the start of text, then type `AI ` to indicate that it has to use [AI plugin](https://github.com/ShamanicArts/Flow.Launcher.Plugin.AI-Assistant). Then AI will create it for you, you use it wherever you want.

#### 2: Proofread with AI
Suppose you need to proofread a text in a document you are writing. Select the text, press `Alt+Shift+Space`, press `Home` to get to the start of the text, write `AI proofread: ` and voila!

Many other examples are possible. mostly with AI stuff and text manipulation. for example- [using a dictionary](https://github.com/tenmo2003/Flow.Launcher.Plugin.FreeDictionary), [translating text](https://github.com/ralosant/FlowLauncher.Google.Translator), [generating a QR code](https://github.com/asmpro7/QrFlow), etc.

**Reply by onesounds** [reply]
![image](https://github.com/user-attachments/assets/94dd6464-cc9e-44c1-84c4-ff57fc21a24b)

As far as I know, this feature was implemented a long time ago.
Could you try configuring it like this and see if the copied text gets pasted properly?
